scripts/traiage.sh

```diff
@@ -133,34 +115,85 @@ wait_agentapi_stable() {
 		--silent \
 		"${CODER_URL}/api/v2/users/me" | jq -r '.username')
 
+	set +o pipefail
 	for attempt in {1..120}; do
-		response=$(curl \
-			--fail \
+		# First wait for task to start
+		task_status=$("${CODER_BIN}" \
+			--url "${CODER_URL}" \
+			--token "${CODER_SESSION_TOKEN}" \
+			exp tasks status "${TASK_NAME}" \
+			--output json |
+			jq -r '.status')
+		echo "Task status is ${task_status}"
+		if [[ "${task_status}" != "running" ]]; then
+			echo "Waiting for task status to be running (attempt ${attempt}/120)"
+			sleep 5
+			continue
+		fi
+
+		# Workspace agent must be healthy
+		healthy=$("${CODER_BIN}" \
+			--url "${CODER_URL}" \
+			--token "${CODER_SESSION_TOKEN}" \
+			exp tasks status "${TASK_NAME}" \
+			--output json |
+			jq -r '.workspace_agent_health.healthy')
+		if [[ "${healthy}" == "true" ]]; then
+			echo "Workspace agent is healthy"
+		else
+			echo "Workspace agent not yet healthy (attempt ${attempt}/120)"
+			sleep 5
+			continue
+		fi
+
+		# AgentAPI application should not be 404'ing
+		agentapi_app_status_code=$(curl \
+			--header "Content-Type: application/json" \
+			--header "Coder-Session-Token: ${CODER_SESSION_TOKEN}" \
+			--location \
+			--request GET \
+			--show-error \
+			--silent \
+			--output /dev/null \
+			--write-out '%{http_code}' \
+			"${CODER_URL}/@${username}/${TASK_NAME}/apps/${APP_SLUG}/status")
+		echo "Workspace app ${APP_SLUG} returned ${agentapi_app_status_code}"
+		if [[ "${agentapi_app_status_code}" != "200" ]]; then
+			echo "Agentapi not yet running (attempt ${attempt}/120)"
+			sleep 5
+			continue
+		fi
+
+		# AgentAPI must be stable
+		agentapi_status=$(curl \
 			--header "Content-Type: application/json" \
 			--header "Coder-Session-Token: ${CODER_SESSION_TOKEN}" \
 			--location \
 			--request GET \
 			--show-error \
 			--silent \
-			"${CODER_URL}/@${username}/${WORKSPACE_NAME}/apps/agentapi/status" | jq -r '.status')
-		if [[ "${response}" == "stable" ]]; then
+			"${CODER_URL}/@${username}/${TASK_NAME}/apps/${APP_SLUG}/status" | jq -r '.status')
+		if [[ "${agentapi_status}" == "stable" ]]; then
 			echo "AgentAPI stable"
-			break
+		else
+			echo "Waiting for AgentAPI to report stable status (attempt ${attempt}/120)"
+			sleep 5
+			continue
```

Comment:
Unfortunately it looks like we may need another wait here, as agentapi will become stable before it starts processing the task. We probably need to wait until the prompt shows up in the messages endpoint or something.

Reply:
This ended up looking very similar to what we'd want `coder exp task status --watch` to be. Will update in a separate PR.